docs/pages/environment_creation/index.md

```diff
@@ -5,116 +5,329 @@ title: Environment Creation
+and some auxiliary information, depending on whether `return_info` was `True`. We can use the methods `_get_obs`
+and `_get_info` that we implemented earlier for that:
 
 ```python
-import gym
-from gym import error, spaces, utils
-from gym.utils import seeding
+    def reset(self, seed=None, return_info=False, options=None):
+        # We need the following line to seed self.np_random
+        super().reset(seed=seed) 
+        
+        # Choose the agent's location uniformly at random
+        self._agent_location = self.np_random.integers(0, self.size, size=2) 
 
-class FooExtraHardEnv(gym.Env):
-    metadata = {'render.modes': ['human']}
+        # We will sample the target's location randomly until it does not coincide with the agent's location
+        self._target_location = self._agent_location
+        while np.array_equal(self._target_location, self._agent_location):
+            self._target_location = self.np_random.randint(0, self.size, size=2)
 
-    def __init__(self):
-        # TODO
-        ...
+        observation = self._get_obs()
+        info = self._get_info()
+        return (observation, info) if return_info else observation
+```
+
+### Step
+The `step` method usually contains most of the logic of your environment. It accepts an `action`, computes the state of 
+the environment after applying that action and returns the 4-tuple `(observation, reward, done, info)`.
+Once the new state of the environment has been computed, we can check whether it is a terminal state and we set `done`
+accordingly. Since we are using sparse binary rewards in `GridWorldEnv`, computing `reward` is trivial once we know `done`. To gather
+`observation` and `info`, we can again make use of `_get_obs` and `_get_info`:
+
+```python
     def step(self, action):
-        # TODO
-        ...
-    def reset(self):
-        # TODO
-        ...
-    def render(self, mode='human'):
-        # TODO
-        ...
+        # Map the action (element of {0,1,2,3}) to the direction we walk in
+        direction = self._action_to_direction[action]
+        # We use `np.clip` to make sure we don't leave the grid
+        self._agent_location = np.clip(
+            self._agent_location + direction, 0, self.size - 1
+        )
+        # An episode is done iff the agent has reached the target
+        done = np.array_equal(self._agent_location, self._target_location)
+        reward = 1 if done else 0       # Binary sparse rewards
+        observation = self._get_obs()
+        info = self._get_info()
+
+        return observation, reward, done, info
+```
+
+### Rendering
+Here, we are using PyGame for rendering. A similar approach to rendering is used in many environments that are included
+with Gym and you can use it as a skeleton for your own environments:
+
+```python
+    def render(self, mode="human"):
+        if self.window is None and mode == "human":
+            pygame.init()
```

Comment:
I think there's a missing import for pygame, it can be done lazily here (and in `env.close`)

Reply:
In the gym-examples repo, I am importing everything at the very beginning. I know that pygame is being imported within `render` in Gym, but imo that's not really necessary for custom environments and might even be a little confusing. 
But I added the imports to the code in the "Declaration and Initialization" section.